app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,10 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    friends_ids = current_user.friends.pluck(:id).push(current_user.id)
+    friends_posts = Post.where(user_id: friends_ids).ordered_by_most_recent
+
+    @timeline_posts ||= friends_posts
   end
```

Comment:
You can use another method in the post model to check for the `current_user` and friend's posts.

`
```ruby
def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
````